azure-pipelines/end-to-end-tests-dir/asset-caching.ps1

```diff
@@ -69,159 +80,252 @@ if (-not ($actual -match $expected)) {
-"A suitable version of .* was not found \(required v[0-9\.]+\)."
-"Asset cache hit for .*; downloaded from: .*"
+"A suitable version of cmake was not found \(required v[0-9\.]+\)\.",
+"Trying to download cmake-[0-9.]+-[^.]+\.(zip|tar\.gz) using asset cache file://$assetCacheRegex/[0-9a-z]+",
+"Download successful! Asset cache hit, did not try authoritative source https://github\.com/Kitware/CMake/releases/download/[^ ]+"
 ) -join "`n"
 
 if (-not ($actual -match $expected)) {
     throw "Failure: asset cache hit"
 }
 
-# Testing asset caching && x-block-orgin promises when --debug is passed (enabled)
-Refresh-TestRoot
-$actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("install", "vcpkg-internal-e2e-test-port", "--overlay-ports=$PSScriptRoot/../e2e-ports", "--x-asset-sources=x-azurl,file://$AssetCache,,readwrite;x-block-origin", "--downloads-root=$DownloadsRoot", "--debug"))
-if (-not ($actual.Contains("[DEBUG] External asset downloads are blocked (x-block-origin is enabled)") -and $actual.Contains("[DEBUG] Asset caching is enabled."))) {
-    throw "Failure: couldn't find expected debug promises (asset caching enabled + x-block-origin enabled)"
-}
-
-# Testing asset caching && x-block-orgin promises when --debug is passed (disabled)
-Refresh-TestRoot
-$actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("install", "vcpkg-internal-e2e-test-port", "--overlay-ports=$PSScriptRoot/../e2e-ports", "--x-asset-sources=clear", "--downloads-root=$DownloadsRoot", "--debug"))
-if (-not ($actual.Contains("[DEBUG] External asset downloads are allowed (x-block-origin is disabled)") -and $actual.Contains("[DEBUG] Asset cache is not configured"))) {
-    throw "Failure: couldn't find expected debug promises (asset caching disabled + x-block-origin disabled)"
-}
-
 # azurl (no), x-block-origin (no), asset-cache (n/a), download (fail)
 # Expected: Download failure message, nothing about asset caching
 Refresh-TestRoot
 $actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("x-download", "$downloadsRoot/example3.html", "--sha512", "d06b93c883f8126a04589937a884032df031b05518eed9d433efb6447834df2596aebd500d69b8283e5702d988ed49655ae654c1683c7a4ae58bfa6b92f2b73a", "--url", "https://localhost:1234/foobar.html"))
-if (-not ($actual.Contains("error: https://localhost:1234/foobar.html: curl failed to download with exit code 7"))) {
+Throw-IfNotFailed
+if (-not ($actual -match "curl: \(7\) Failed to connect to localhost port 1234 after \d+ ms: Could not connect to server")) {
     throw "Failure: azurl (no), x-block-origin (no), asset-cache (n/a), download (fail)"
 }
-
 #azurl (no), x-block-origin (no), asset-cache (n/a), download (sha-mismatch)
 #Expected: Download message with the "you might need to configure a proxy" message and with expected/actual sha
 Refresh-TestRoot
+$expected = @(
+"Downloading https://example\.com -> example3\.html",
+"[^\n]+example3\.html\.\d+\.part: error: download from https://example\.com had an unexpected hash",
```

Comment:
Do we want the `error:`? 

```suggestion
"[^\n]+example3\.html\.\d+\.part: warning: download from https://example\.com had an unexpected hash",
```

Reply:
It's fatal and we can't continue after it happens, so I think it needs to be error?

Reply:
No, SHA mismatch immediately dies. We do not try anything else.

Reply:
We did decide to make this an ordinary download failure rather than a fatal error, but this remains 'error'. In the cases where it is like a warning, we print nothing because the overall download succeeded. The error is only printed if there's no authoritative source that can do it either.